src/HybridSolver.cpp

```diff
@@ -300,44 +283,35 @@ void HybridSolver::AssembleHybridSystem(
         }
 
         // Hybrid_el_ = CMinvCT - CMinvDTAinvDMinvCT
-        tmpHybrid_el -= CMDADMC;
+        Hybrid_el_[iAgg] -= CMDADMC;
 
         // Add contribution of the element matrix to the global system
-        HybridSystem_->AddSubMatrix(local_multiplier, local_multiplier,
-                                    tmpHybrid_el);
-
-        // Save element matrix [C 0][M B^T;B 0]^-1[C 0]^T
-        Hybrid_el_[iAgg] = tmpHybrid_el;
+        H_proc.AddSubMatrix(local_multiplier, local_multiplier, Hybrid_el_[iAgg]);
     }
+
+    return H_proc;
 }
 
+
 /// @todo nonzero BC, solve on true dof
```

Comment:
I think this comment might be out of date (aren't you solving on true dof now?)

Reply:
This is a confusing comment (I will update shortly). Now `HybridSolver` works  for nonzero Dirichlet BC, but it still does not work for nonzero Neumann BC (essential edge unknown). In fact, I think even `MinresBlockSolver` doesn't work in the case of nonzero Neumann BC. 

For the second half, the hybridized system is on true dof, but `HybridSolver::Mult` is still assuming that input vectors (on the unknowns of the original problem) are not based on true dof (unlike `MinresBlockSolver`).